source/CCustomOpcodeSystem.cpp

```diff
@@ -652,23 +656,29 @@ namespace CLEO {
 			case DT_LVAR_ARRAY:
 			{
 				GetScriptParams(thread, 1);
-				char* str = opcodeParams[0].pcParam;
 
-				size_t length;
-				if(str != nullptr)
-					length = strlen(str);
-				else
+				size_t length = bufSize ? bufSize - 1 : 0; // minus terminator char
+				
+				if (opcodeParams[0].dwParam == 0)
 				{
-					length = 0;
-					LOG_WARNING("Reading string from null pointer in script %s", ((CCustomScript*)thread)->GetInfoStr().c_str());
+					const char* eStr = "(null)";
+					length = min(length, strlen(eStr));
+					if (length) strncpy(buf, eStr, length);
+				}
+				else if(opcodeParams[0].dwParam <= CCustomOpcodeSystem::MinValidAddress)
+				{
+					LOG_WARNING("Reading string from argument with invalid pointer value (0x%X) in script %s", opcodeParams[0].dwParam, ((CCustomScript*)thread)->GetInfoStr().c_str());
+					const char* eStr = "(badPtr)";
+					length = min(length, strlen(eStr));
+					if (length) strncpy(buf, eStr, length);

```

Comment:
here we ignore a bad input and let script continue working. I think it's better if we had another version of this method, that returned a boolean indicating if the read was successful or not. then gradually update existing opcodes to use the new method and suspend if the read failed.

Reply:
It might return nullptr on fail.

Reply:
Added error checks and script suspending.